packages/appframe/src/react/index.js

```diff
@@ -0,0 +1,211 @@
+    setSidenavOverlayed(props.sidenavOverlayed)
+  }, [props.sidenavOverlayed])
+
+  const getVariant = useCallback(() => {
+    const { sidenavVariants: variants } = vars
+
+    if (!sidenav) return variants.closed
+
+    if (sidenavOverlayed) return variants.overlay
+
+    if (sidenavOpen) {
+      return mediumMedia ? variants.open : variants.overlay
+    } else {
+      return mediumMedia ? variants.minimized : variants.closed
+    }
+  }, [sidenav, sidenavOpen, sidenavOverlayed, mediumMedia])
+
+  const [variant, setVariant] = useState(getVariant)
+  useEffect(() => {
+    setVariant(getVariant)
+  }, [getVariant])
+
+  const skipTargetId = 'ps-frame-skip-target'
+  const skipTargetRef = useRef()
+
+  const focusSkipTarget = useCallback(() => {
+    if (!skipTargetRef.current) return
+
+    skipTargetRef.current.focus()
+  }, [])
+
+  return (
+    <div
+      ref={ref}
+      {...styles.appframe(themeName, props)}
+      {...filterReactProps(props)}
+    >
+      <Theme name={themes.dark}>
+        <SkipBanner href={'#' + skipTargetId} />
+      </Theme>
+
+      <Theme name={themes.dark}>
+        <div {...styles.topnav()}>{topnav}</div>
+      </Theme>
+
+      <Container variant={variant}>
+        {sidenav && <SideNav variant={variant}>{sidenav}</SideNav>}
+
+        <main {...styles.content()}>
+          <SkipTarget
+            id={skipTargetId}
+            onClick={focusSkipTarget}
+            ref={skipTargetRef}
+          />
+
+          {children}
+        </main>
+      </Container>
+    </div>
+  )
+})
+
+AppFrame.displayName = 'AppFrame'
+AppFrame.defaultProps = {
+  sidenavOpen: false,
+  sidenavOverlayed: false
+}
+
+AppFrame.propTypes = {
+  children: PropTypes.oneOfType([PropTypes.func, PropTypes.node]),
+  sidenav: PropTypes.oneOfType([PropTypes.func, PropTypes.node]),
+  sidenavOpen: PropTypes.bool,
+  sidenavOverlayed: PropTypes.bool,
+  topnav: PropTypes.oneOfType([PropTypes.func, PropTypes.node])
+}
+
+function SkipBanner(props) {
+  return (
+    <div {...styles.skipBanner()} {...props}>
+      <Button
+        appearance={Button.appearances.secondary}
+        size={Button.sizes.small}
+        tabIndex={0}
+      >
+        skip to main content
+      </Button>
+    </div>
+  )
+}
+SkipBanner.propTypes = {
+  href: PropTypes.string.isRequired
+}
+
+const SkipTarget = React.forwardRef((props, ref) => {
+  return <a ref={ref} tabIndex={-1} {...props} />
+})
+SkipTarget.displayName = 'SkipTarget'
+SkipTarget.propTypes = {
+  id: PropTypes.string.isRequired
+}
+
+function Container(props) {
+  const { variant, ...rest } = props
+
+  return <div {...styles.container(variant)} {...rest} />
+}
+Container.propTypes = {
+  children: PropTypes.node,
+  variant: PropTypes.oneOf(Object.keys(vars.sidenavVariants))
+}
+
+function SideNav(props) {
+  const { children, variant, ...rest } = props
+
+  const ref = React.useRef()
+
+  const lockScroll = variant === vars.sidenavVariants.overlay
+  useBodyScrollLock(ref, lockScroll)
```

Comment:
This prevents the body from scrolling right? Is only until you've scrolled to the end of the sidebar or all the time? Would a e.preventDefault() option on Scrollable have done the same?

Reply:
not on Scrollable but maybe intercepting all scroll events and preventing those. i seem to remember that you do that for some reason so i've always do it this way. also this isn't testable in storybook - i verified it manually